src/classes/BookingSlotUtilities.cls

```diff
@@ -0,0 +1,233 @@
+			for(Booking_Slot__c bookingSlot : newBookingSlotList) {
+				Booking_Slot__c oldBookingSlot = oldBookingSlotMap.get(bookingSlot.Id);
+				if(bookingSlot.Driver_Availability__c != oldBookingSlot.Driver_Availability__c) {
+					if(bookingSlot.Driver_Availability__c != null) {
+						driverAvailabilityToRecalculate.add(bookingSlot.Driver_Availability__c);
+					}
+					if(oldBookingSlot.Driver_Availability__c != null) {
+						driverAvailabilityToRecalculate.add(oldBookingSlot.Driver_Availability__c);
+					}
+				} 
+			}
+		} else {
+			for(Booking_Slot__c bookingSlot : newBookingSlotList) {
+				if(bookingSlot.Driver_Availability__c != null) {
+					driverAvailabilityToRecalculate.add(bookingSlot.Driver_Availability__c);
+				}
+			}
+		}
+
+		return driverAvailabilityToRecalculate;
+
+
+	}
+
+	private List<Booking_Slot__c> filterBookingSlotsWithChangedDriver (Map<Id, Booking_Slot__c> oldBookingSlotMap, List<Booking_Slot__c> newBookingSlotList) {
+
+		List<Booking_Slot__c> bookingSlotsToUpdate = new List<Booking_Slot__c>();
+
+		if(oldBookingSlotMap != null) {
+			for(Booking_Slot__c bookingSlot : newBookingSlotList) {
+				Booking_Slot__c oldBookingSlot = oldBookingSlotMap.get(bookingSlot.Id);
+				if(driverAndDateNotNull(bookingSlot) && driverOrDateHasChanged(oldBookingSlot, bookingSlot)) {
+					bookingSlotsToUpdate.add(bookingSlot);
+				}
+			}
+		} else {
+			for(Booking_Slot__c bookingSlot : newBookingSlotList) {
+				if(driverAndDateNotNull(bookingSlot)) {
+					bookingSlotsToUpdate.add(bookingSlot);
+				}
+			}
+		}
+
+		return bookingSlotsToUpdate;
+
+
+	}
+
+
+	private Set<Id> getDriverIds (List<Booking_Slot__c> bookingSlotList) {
+
+		Set<Id> driverIdSet = new Set<Id>();
+
+		for(Booking_Slot__c bookingSlot : bookingSlotList) {
+			driverIdSet.add(bookingSlot.Driver__c);
+		}
+
+		return driverIdSet;
+	}
+
+
+
+
+	private void assignDriverAsOwner(List<Booking_Slot__c> bookingSlotList) {
+
+		mapOfDriverIdToUserId = getUserLinkedToDriver(bookingSlotList);
+
+		for(Booking_Slot__c bookingSlot : bookingSlotList) {
+			if(driverIsLinkedToAUser(bookingSlot)) {
+				bookingSlot.OwnerId = mapOfDriverIdToUserId.get(bookingSlot.Driver__c);
+			}
+		}
+
+	}
+
+
+
+	private Map<Id, Id> getUserLinkedToDriver(List<Booking_Slot__c> bookingSlots) {
+
+		Set<Id> driverIds = new Set<Id>();
+		Map<Id, Id> mapOfDriverIdToUserId = new Map<Id, Id>();
+
+		for(Booking_Slot__c bookingSlot : bookingSlots) {
+			driverIds.add(bookingSlot.Driver__c);
+		}
+
+		List<Driver__c> drivers = [SELECT Id, User__c FROM Driver__c WHERE Id IN :driverIds];
+
+		for(Driver__c driver : drivers) {
+			mapOfDriverIdToUserId.put(driver.Id, driver.User__c);
+		}
+
+
+		return mapOfDriverIdToUserId;
+	}
+
+	private Boolean ownerIsNotCurrentlyDriver(Booking_Slot__c bookingSlot) {
+		return bookingSlot.Driver__c != null && bookingSlot.Driver__c != bookingSlot.OwnerId;
+
+	}
+
+	private Boolean driverIsLinkedToAUser(Booking_Slot__c bookingSlot) {
```

Comment:
Methods should use the `lowerCamelCase` naming convention. Please rename accordingly.